pkg/api/repository.go

```diff
@@ -36,20 +36,12 @@ func (r *RepositoryRequest) FillDefaults() {
 	//Fill in default values in case of PUT request, doesn't have to be valid, let the db validate that
 	defaultName := ""
 	defaultUrl := ""
-	defaultVersions := []string{}
-	defaultArch := ""
 	if r.Name == nil {
 		r.Name = &defaultName
 	}
 	if r.URL == nil {
 		r.URL = &defaultUrl
 	}
-	if r.DistributionVersions == nil {
-		r.DistributionVersions = &defaultVersions
-	}
-	if r.DistributionArch == nil {
-		r.DistributionArch = &defaultArch
-	}
```

Comment:
This is still necessary

For example:
- I have a repository, `foo` with arch `x86_64`. 
- I PUT update `foo` with just `{ "name": "new foo"}`
- I expect the arch of foo to become `any`, but it's still `x86_64`.

You should also add a test to cover this scenario

Reply:
This makes sense, but I think a bigger issue with the put update you describe is that the name/url are not required. If this is truly a PUT, you should likely require a set of variables no? What would happen to the default URL in such a situation? Should the version array switch to the default? The existing GPGkey? and validation setting (that is still to come)?

Reply:
Name/URL are essentially required. The default URL/name is `""`, and the API will return an error because validation will catch that, like here: https://github.com/content-services/content-sources-backend/blob/main/pkg/models/repository_configuration.go#L78. Perhaps they should be marked as required in the API spec, if that's what you mean?

Version array should default to `["any"]`, just like it does for create. 

gpgKey and verification default to `""` and `false`.

The idea is to reset whatever values are specified in the PUT request. The body is supposed to be a complete resource, so anything missing gets its default value.

Reply:
Got it. I'll reset these to their previous values.